packages/autorest.go/test/autorest/paginggroup/zz_options.go

```
@@ -16,7 +16,8 @@ type CustomParameterGroup struct {
 
 // PagingClientAppendAPIVersionOptions contains the optional parameters for the PagingClient.NewAppendAPIVersionPager method.
 type PagingClientAppendAPIVersionOptions struct {
-	// placeholder for future optional parameters
+	// Resumes the paging operation from the provided link.
+	NextLink string
```

We'll need to ensure that this value corresponds to the `client.method`. And since we're supposed to treat next links as opaque values, I don't think we can simply parse the provided value to ensure that the host/path matches (while host should _always_ match, it's possible that path will not).

We have the same constraints with `Poller[T]`. The way we solved it there is we have method `Poller[T].ResumeToken()` which essentially dumps the data structure to JSON. So, when you create a new `Poller[T]` from the resume token, if it doesn't match an error is returned.

While a "next link token" (like above) would solve this, it does mean that you can't just take an arbitrary next link value. You'd have to create the token first from the `Pager[T]` (mock-up below). How important do you think it is that we accept a "naked" next link?

```go
pager := client.NewFooPager(nil)
tk := pager.ResumeToken()
// other code...
pager = client.NewFooPager(&NewFooPagerOptions{
    ResumeToken: tk,
})
```

---

Hmm, this poses a problem though.
```go
pager := client.NewFooPager(nil)
tk := pager.ResumeToken()
pager = client.NewBarPager(&NewBarPagerOptions{
    ResumeToken: tk,
})
```
Creating a `Pager[Bar]` from a `Pager[Foo]` token should return an error. However, our `New*Pager` methods don't return an error. :(

---

All of this was why I was asking in the e-mail thread about asking folks to update their public-facing typespec to actually encode which query parameter is being used for pagination record-keeping. We also shouldn't allow the domain to change, or the apiVersion, etc. Not sure we can do anything about this if all we have to work with is an opaque value.

Regarding a `.ResumeToken()` getter - if we keep with the opaque value, we can't parse it to determine anything, so the best we could do is to ensure that the type (e.g. a KeyVault Secret) is the same by saving some tagged union `{"tag":"KeyVault.Secret","nextLink":"..."}`. I would imagine that this is the least likely mistake anyone would make when reading such a token from durable storage on process start, when resuming from some prior list.

The big downside I see there is that `NextLink` is already exported from the page types, so it would not be clear to users of the SDK what the difference is and why to use one over the other, or, if all you could use were the typed/encoded one, why the "raw" `NextLink` is exported.